### DISCUSSION TITLE
problem with hatches and autodesk products + workaround

### DISCUSSION THREAD

**Discussion by oattila** [tarikjabiri/dxf#86] [General]
I ran into a problem with exported DXFs with hatches. Autodesk products (DWG TrueView for example) could not open the files, while some other software could. After a lot of investigation I found out that what Autodesk doesn't like is if extrusion properties in hatch come before the elevation, which happens when using dxfjs writer. I modified the writer so that extrusion is dxfied after elevation, which fixed the problem.

**Comment by tarikjabiri** [comment]
Yeah the extrusion shouldn't be in entity class we should add it directly to the entities that support it and in the right order I will try to fix it.
